### PR TITLE
Add state transaction api to spec

### PR DESCRIPTION
**Pull request by youngbupark**
## Description

Add transaction API

## Issue reference

#749

### CODE REVIEW

**Comment by youngbupark** [comment]
@orizohar @yaron2 @amanbha @LMWF  please review this doc change.

File: reference/api/state_api.md [hunk 1]
```diff
@@ -274,6 +275,79 @@ None.
...
+## State transactions
+
+Persists the changed to the state for an actor as a multi-item transaction.
+
+***Note that this operation is dependant on a using state store component that supports multi-item transactions.***
+
+#### HTTP Request
+
+```http
+POST/PUT http://localhost:<daprPort>/v1.0/v1.0/state/<storename>/transaction
```
Comment (LMWF) [inline reference/api/state_api.md:287]: Extra v1.0

File: reference/api/state_api.md [hunk 2]
```diff
@@ -274,6 +275,79 @@ None.
 curl -X "DELETE" http://localhost:3500/v1.0/state/starwars/planet -H "ETag: xxxxxxx"
 ```
 
+## State transactions
+
+Persists the changed to the state for an actor as a multi-item transaction.
```
Comment (LMWF) [inline reference/api/state_api.md:280]: changed -> changes

Comment (LMWF) [inline reference/api/state_api.md]: "for an actor" - this is not just for actors anymore

File: reference/api/state_api.md (hunk 1 above)
Comment (youngbupark) [reply reference/api/state_api.md]: fixed.

File: reference/api/state_api.md (hunk 2 above)
Comment (youngbupark) [reply reference/api/state_api.md]: fixed.

Comment (youngbupark) [reply reference/api/state_api.md]: removed :)

File: reference/api/state_api.md [hunk 3]
```diff
@@ -274,6 +275,79 @@ None.
 curl -X "DELETE" http://localhost:3500/v1.0/state/starwars/planet -H "ETag: xxxxxxx"
 ```
 
+## State transactions
+
+Persists the changes to the state store as a multi-item transaction.
+
+***Note that this operation is dependant on a using state store component that supports multi-item transactions.***
```
Comment (yaron2) [inline reference/api/state_api.md:282]: Unless this is listed somewhere else, and I don't think this is, we need to put a list here of supported state stores.

Comment (LMWF) [reply reference/api/state_api.md:282]: There is a kind of a list here, but it's in the context of actors.  Yeah there should be one for "state stores that support multi-item transactions" https://github.com/dapr/docs/blob/master/concepts/actors/actors_features.md#actor-state-management

Comment (yaron2) [reply reference/api/state_api.md:282]: @youngbupark can you add the following list here as transactional state stores?

* Redis
* MongoDB
* PostgreSQL
* SQL Server
* Azure CosmosDB

Comment (yaron2) [reply reference/api/state_api.md:282]: Added

**Pull request merged**